### ISSUE TITLE
Why isn't `Q1.7` a valid specifier?

### ISSUE THREAD

**Issue by metasim** [xpe/q-num#1]
Thanks for the great library! Exactly what I needed!

I have an application using an ADC where the documentation _claims_ the I/Q samples are each 8-bit "Q7" fixed point values, used to represent values in `[-1.0, 1.0)`.  (The documentation is a bit confusing and perhaps even inconsistent, and I'm new to this domain).

Using ARM notation, I'm assuming this means `Q1.7`, but when I attempt this with `q-num`, I get the following error

```
error: proc macro panicked
   --> txrx.rs:134:1
    |
134 | define_q_num!(Q7, Q1.7);
    | ^^^^^^^^^^^^^^^^^^^^^^^
    |
    = help: message: assertion failed: input.int_bits >= 2
```

Should this be a valid encoding, or am I misunderstanding something?

**Comment by xpe** [comment]
Hello. Thanks for your feedback. You are correct to expect that `Q1.7` would work.

I've found the issue in the code. For some (incorrect) reason (or lack of reasoning 🫣), when I wrote the code, I included this check: `assert!(input.int_bits >= 2);`. 🤷  I'm fixing this and adding some tests now.

**Comment by xpe** [comment]
Ok, fixed.

> I have an application using an ADC where the documentation claims the I/Q samples are each 8-bit "Q7" fixed point values, used to represent values in [-1.0, 1.0). (The documentation is a bit confusing and perhaps even inconsistent, and I'm new to this domain).

This looks correct to me. As I've implemented it (according to the standard), the range of `Q1.7` is [-1.0, 0.9921875] with a step size of 0.0078125.

**Comment by xpe** [comment]
Fix available in v0.1.1 : https://github.com/xpe/q-num/releases/tag/v0.1.1 and https://crates.io/crates/q-num/0.1.1

**Comment by metasim** [comment]
@xpe Thanks so much! So glad you are able to fix it! Counting bits is surprisingly hard at times 😆
